### DISCUSSION TITLE
OutboundConfigurationObject 中的ServerObject能配置指定的DNS吗

### DISCUSSION THREAD

**Discussion by snail2sky** [v2fly/v2ray-core#1906] [Q&A] [answered]
- 我有公网IP，但是公网IP不定时的会变，所以，我用DDNS，当ip改变时，就把域名同步更新到godaddy域名服务商
- 但是，正式因为这一点，可能v2ray 或者 我本机记住了变更前的ip，所以导致连接到错误的ip
- OutboundConfigurationObject 中的ServerObject 配置的 address 就是我的域名，这就很难受，本以为可以利用这一点
- 来解决掉我公网ip不稳定的情况，但是由于缓存，导致即使域名能指向真是服务器的Ip，但客户端最终因为缓存而无法找到真实服务器ip
- 求大佬们给点建议，看看我这种情况怎么搞会比较好

**Comment by kingwilliam** [comment, marked as answer]
首先 您要认识dns内每笔资料都有一个TTL设定, 全名是 time to live, 用途就是通知每个客户这个纪录有效时间是多少(单位通常是"秒")

假如您的 www.example.com 在ddns最後更新时间是 00:00:01 零时零分01秒 TTL 是 300(即5分钟)

如 客户A 在 00:00:02 查询, 在每级的dns, 包括您公司的wifi router或您最终的手机也会纪录
在 00:00:02 - 00:05:02 都是有效, 在有效时间内, 不会再向上级dns查询 直至超时 
如 00:05:03或以後, 如再次查询 因之前的纪录已超时, 便会向 上级dns再查询.

但 如客户B 在 00:04:59 查询, 他的有效时间便会是 00:04:59 - 00:09:59

但 如您的 www.example.com 在ddns最後更新时间是 00:04:30
这时 客户Ａ因还没更新到最新 IP, 所以在 00:04:30 - 00:05:01 是会连接不到。 直至 00:05:02重新向上级dns再查询才会回复正常。
这就是你所说的 缓存.
就算您的手机或您上级的wifi router dns都取消缓存也解决不了您的请况, 因您的dns服务公司也有他们的dns缓存.

解说完， 如何解决？ 
完全解决可以是有 也可以是没有。

如您有多於一台伺服器做 load balancing, 问题就可解决

但如只得一台 只能在ddns内的ttl尽可能小(但一般ddns最小通常都是300s或180s)
这也有不能连接时间。 只能尽量可能小.

**Reply by snail2sky** [reply]
好的 明白了，感谢回答

**Comment by Emmricjordans** [comment]
Zkd!kdmr!d!!r!kdkdf!vlr??xla?a?lx!d!dlc!kkrkujdejen
Kmjxneunrnux!!jdljsd!lodlr!due
